server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sServerToHttpInterpreter.scala

```diff
@@ -65,8 +66,8 @@ trait Http4sServerToHttpInterpreter[F[_], G[_]] {
       )
 
       OptionT(interpreter(serverRequest, serverEndpoints).flatMap {
-        case None           => none.pure[G]
-        case Some(response) => fToG(serverResponseToHttp4s(response)).map(_.some)
+        case _: ServerInterpreterResult.Failure        => none.pure[G]
```

Comment:
I'm struggling a bit here to surface the `MethodNotAllowed` here, as I'm very unfamiliar with http4s. It currently does not past the 405 test I added to `ServerBasicTests`. I'll keep plugging away, but suggestions welcome.

Reply:
This seems like it might be an unsolved problem with http4s: https://github.com/http4s/http4s/issues/234. Going to punt on it for now, and only test the servers where returning 405 is possible.

Play also seems like a bit of a challenge, so removed the test from `PlayServerTest` as well.